imcsdk/apis/admin/ldap.py

```diff
@@ -0,0 +1,549 @@
+        'domain': domain,
+        'encryption': ('disabled', 'enabled')[encryption],
+        'timeout': str(timeout),
+        'user_search_precedence': user_search_precedence,
+        'bind_method': bind_method,
+        'bind_dn': bind_dn,
+        'password': password,
+        'filter': filter,
+        'attribute': attribute,
+        'group_attribute': group_attribute,
+        'group_nested_search': str(group_nested_search),
+        'group_auth': ('disabled', 'enabled')[group_auth],
+        'locate_directory_using_dns': ('no', 'yes')[locate_directory_using_dns],
+        'dns_domain_source': dns_domain_source,
+        'dns_search_domain': dns_search_domain,
+        'dns_search_forest': dns_search_forest
+        }
+
+    mo.set_prop_multiple(**params)
+
+    if ldap_servers:
+        _set_ldap_servers(mo, ldap_servers)
+
+    mo.set_prop_multiple(**kwargs)
+    handle.set_mo(mo)
+    return handle.query_dn(mo.dn)
+
+
+def _check_match(prop1, prop2):
+    if _is_invalid_value(prop1) and _is_invalid_value(prop2):
+        return None
+    return prop1 == prop2
+
+
+def _check_ldap_server_match(ldap_mo, ldap_servers):
+    mo = AaaLdap(parent_mo_or_dn="sys")
+    _set_ldap_servers(mo, ldap_servers)
+    for server, port in zip(_LDAP_SERVERS, _LDAP_SERVER_PORTS):
+        conf_server = getattr(ldap_mo, server)
+        in_server = getattr(mo, server)
+
+        match = _check_match(conf_server, in_server)
+        # IMC assigns a default value of port even when a valid server ip is
+        # not assigned. The following logic will skip checking for ports where
+        # the server ip itself is not specified, but default port is set.
+        if match is None:
+            continue
+
+        if not match:
+            return False
+
+        conf_port = getattr(ldap_mo, port)
+        in_port = getattr(mo, port)
+
+        if _check_match(conf_port, in_port) is False:
+            return False
+
+    return True
+
+
+def _check_ldap_specials(mo, kwargs):
+    params = {}
+    if _is_valid_arg('enabled', kwargs):
+        params['admin_state'] = ('disabled', 'enabled')[kwargs.pop('enabled')]
+
+    if _is_valid_arg('encryption', kwargs):
+        params['encryption'] = ('disabled', 'enabled')[kwargs.pop('encryption')]
+
+    if _is_valid_arg('group_auth', kwargs):
+        params['group_auth'] = ('disabled', 'enabled')[kwargs.pop('group_auth')]
+
+    if _is_valid_arg('locate_directory_using_dns', kwargs):
+        params['locate_directory_using_dns'] = ('no', 'yes')[kwargs.pop('locate_directory_using_dns')]
+
+    if _is_valid_arg('timeout', kwargs):
+        params['timeout'] = str(kwargs.pop('timeout'))
+
+    if _is_valid_arg('group_nested_search', kwargs):
+        params['group_nested_search'] = str(kwargs.pop('group_nested_search'))
+
+    if not mo.check_prop_match(**params):
+        return False
+
+    # pop the password property if it exists
+    if _is_valid_arg('password', kwargs):
+        kwargs.pop('password')
```

Comment:
Guess, I spoke to soon in the code review process. you next function sort of makes it clear, why you need it. :-)